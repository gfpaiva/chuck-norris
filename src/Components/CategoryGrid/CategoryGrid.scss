@import '../../Scss/root';

.category-grid {
  &__item {
    transition: transform .25s ease;

    &--active { background-color: $success-color }
    &:hover { transform: scale(1.05) }
  }

  &__icon {
    @include rfs($fs-big);
    display: inline-block;
    margin-bottom: 1rem;
  }

  &__name {
    @include rfs($fs-medium);
    font-weight: 100;
  }
}
