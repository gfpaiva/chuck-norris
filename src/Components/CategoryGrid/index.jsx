import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import Card from '../Card';
import Grid from '../Grid';

import { getCategory } from '../../Utils/Categories';

import './CategoryGrid.scss';

function CategoryGrid({ active, categories }) {
  return (
    <div className="category-grid">
      {categories && !!categories.length && (
        <Grid>
            {
              categories.map((category) => {
                const detailCategory = getCategory(category) ? getCategory(category) : { icon: '', link: `/${category}`, content: category.toUpperCase };

                const { icon, content, link } = detailCategory;

                return (
                  <Card
                    className={`category-grid__item text-center px-0 py-4${active === category ? ' category-grid__item--active' : ''}`}
                    hover
                    key={content}
                    href={active !== category ? `/category${link}` : ''}
                  >
                    <span className="category-grid__icon">{icon}</span>
                    <br />
                    <span className="category-grid__name">{content}</span>
                  </Card>
                );
              })
            }
        </Grid>
      )}
    </div>
  );
}

CategoryGrid.defaultProps = {
  active: '',
};

CategoryGrid.propTypes = {
  active: PropTypes.string,
  categories: PropTypes.arrayOf(PropTypes.string).isRequired,
};

const mapStateToProps = ({ facts: { categories } }) => ({ categories });

export default connect(mapStateToProps)(CategoryGrid);
