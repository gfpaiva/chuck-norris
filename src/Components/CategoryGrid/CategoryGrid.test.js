import React from 'react';
import wait from 'waait';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';

import store from '../../Store';

import CategoryGrid from './index';

import getCategories from '../../../__mocks__/getCategories';

describe('<CategoryGrid />', () => {
  it('should mount properly', async () => {
    fetch
      .once(JSON.stringify(getCategories));

    const wrapper = mount(
      <Provider store={store}>
        <Router>
          <CategoryGrid />
        </Router>
      </Provider>
    );

    await wait();
    wrapper.update();

    expect(wrapper.find('CategoryGrid')).toMatchSnapshot();
  });

  it('should active category card with className', async () => {
    fetch
      .once(JSON.stringify(getCategories));

    const wrapper = mount(
      <Provider store={store}>
        <Router>
          <CategoryGrid active="dev" />
        </Router>
      </Provider>
    );

    await wait();
    wrapper.update();

    console.log('TCL: wrapper', wrapper.debug());

    expect(wrapper.find('.card.category-grid__item--active')).toHaveLength(1);
    expect(wrapper.find('.card.category-grid__item--active .category-grid__name').text()).toBe('DEV');
  });
});
