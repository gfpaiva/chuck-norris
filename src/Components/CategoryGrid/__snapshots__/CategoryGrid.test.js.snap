// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryGrid /> should mount properly 1`] = `
<CategoryGrid
  active=""
  categories={Array []}
  dispatch={[Function]}
>
  <div
    className="category-grid"
  />
</CategoryGrid>
`;
